### PR TITLE
Migrate to Java8

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR migrates the library to use Java Time (addressing #21).

According to this move, we eliminate all custom time calculation routines, providing conversion methods to corresponding Java Time values.

### CODE REVIEW

**Comment by alexander-yevsyukov** [comment]
@armiol, PTAL.

**Comment by alexander-yevsyukov** [comment]
Further work is needed to improve test coverage. See #24.

File: time/src/test/java/io/spine/time/string/TimestampStringifierShould.java [hunk 1]
```diff
@@ -31,6 +33,9 @@
  */
 public class TimestampStringifierShould extends AbstractStringifierTest<Timestamp> {
 
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
```
Comment (armiol) [inline time/src/test/java/io/spine/time/string/TimestampStringifierShould.java]: As long as we agree on the fact that this rule is deprecated, please consider using `assertThrows` in the test code below instead.

**Review by armiol — APPROVED** [review]
@alexander-yevsyukov LGTM with a single comment to address.

**Pull request merged**